python/cudf/cudf/core/buffer/exposure_tracked_buffer.py

```diff
@@ -139,103 +41,50 @@ def mark_exposed(self) -> None:
         self._exposed = True
 
     @classmethod
-    def _from_device_memory(cls, data: Any, *, exposed: bool = False) -> Self:
-        """Create an exposure tracked buffer from device memory.
-
-        No data is being copied.
-
-        Parameters
-        ----------
-        data : device-buffer-like
-            An object implementing the CUDA Array Interface.
-        exposed : bool, optional
-            Mark the buffer as permanently exposed.
-
-        Returns
-        -------
-        ExposureTrackedBuffer
-            Buffer representing the same device memory as `data`
-        """
-        ret = super()._from_device_memory(data)
+    def _from_device_memory(cls, data: Any, exposed: bool) -> Self:
+        ret = super()._from_device_memory(data, exposed=exposed)
         ret._exposed = exposed
```

Comment:
**issue:** We pass `exposed` to the superclass constructor but are none-the-less on the hook to remember it on the returned object. This feels wrong to me. Either the superclass doesn't know about exposed at all (in which case `_from_device_memory` can't be a shared thing with differing signatures). Or else we should store `exposed` in the superclass and implement the `exposed` property and `mark_exposed` with overrides here.

WDYT?

As it is, I think an `ExposureTrackedBufferOwner` can sit in for any `BufferOwner`, but not vice-versa (perhaps that is intended).

In some senses to be "generic", the return value of `_from_device_memory` is only ever allowed to claim that it supports the interface that `BufferOwner` provides (not that of a subclass)